packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,34 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const currentCardIndex = Array.from(this.children).indexOf(target as HTMLElement);
+
+    // reset to first page if tabbing back into the carousel after previously moving pages
+    if (currentContains && currentCardIndex === 0) {
+      this.start = 0;
+    }
+
+    if (currentContains) {
+      // going forwards, change page depending on card index
+      if (currentCardIndex >= this.start + this.pageSize) {
+        this.start = Math.min(currentCardIndex, this._total + 1 - this.pageSize);
```

Comment:
I think both of our approaches are off by one in different cases. Using the suggested calculation and focusing on the 5th (index `4`) would make the page start with that same card, instead of the 4th (index `3`) as it should. This is because we're only focusing on the last page scenario (subtracting from `this.total`).

Using `currentCardIndex - (currentCardIndex % this.pageSize)` will always ensure getting the right start index for the current card's page.